src/Action/Security/RefreshAccessTokenAction.php

```diff
@@ -76,10 +76,13 @@ public function __construct(
         $this->accessTokenTtl = $jwtAccessTokenTtl;
     }
 
-    public function __invoke(Request $request, Token $refreshToken): Response
+    public function __invoke(Request $request, Token $refreshToken): JsonResponse
     {
         try {
-            $user = $this->userRepository->findByUsernameWithAssignments($refreshToken->getClaim('aud'));
+            assert($refreshToken instanceof Token\Plain);
```

Comment:
Assertions are primary made for debug/develop/test when you want `run or die` scenario.  For business logic is much better to use `if/else`, `throw`, etc. Please replace them.